apps/backoffice/src/app/api/institutions/[institutionId]/groups/route.ts

```diff
@@ -62,3 +52,14 @@ export const GET = withJWTAuthHandler(
     }
   },
 );
+
+const getQueryParam = (
+  request: NextRequest,
+  param: string,
+  defaultValue: number,
+): number => {
+  const rawParam = NumberFromString.decode(
+    request.nextUrl.searchParams.get(param),
+  );
+  return E.isRight(rawParam) ? rawParam.right : defaultValue;
+};
```

Comment:
**chore:** improve validation

With the current implementation, if the queryParam is provided but it is not a valid number (i.e.: `size=foo`), it will ignore the error and continue with default values; but in that case we should return a Bad Request

Reply:
What do you think about a generic method to use when needed to retrieve and validate a queryParam?
Somenthin like this:

```typescript
const extractQueryParam = <A>(
  request: NextRequest,
  param: string,
  decoder: t.Type<A>,
  defaultValue?: A,
): E.Either<Error, A> =>
  pipe(
    request.nextUrl.searchParams.get(param),
    O.fromNullable,
    O.foldW(
      () =>
        defaultValue
          ? E.right(defaultValue)
          : E.left(new Error(`Missing '${param}' parameter in request`)),
      (rawParam) =>
        pipe(
          rawParam,
          decoder.decode,
          E.mapLeft((errors) => E.toError(readableReport(errors))),
        ),
    ),
  );

```

Reply:
We can also define an Error for bad request and intercept it into a try cath that wrap the controller.

Reply:
Ok! seems legit, we can create a dedicated task and address this later on!